test/jdk/java/awt/print/Dialog/DialogType.java

```diff
@@ -21,53 +21,66 @@
  * questions.
  */
 
-/**
+import java.awt.print.PrinterJob;
+import java.lang.reflect.InvocationTargetException;
+import javax.print.attribute.HashPrintRequestAttributeSet;
+import javax.print.attribute.PrintRequestAttributeSet;
+import javax.print.attribute.standard.DialogTypeSelection;
+import jtreg.SkippedException;
+
+/*
  * @test
  * @bug 6568874
+ * @key printer
+ * @library /java/awt/regtesthelpers
+ * @library /test/lib
+ * @build PassFailJFrame
+ * @build jtreg.SkippedException
  * @summary Verify the native dialog works with attribute sets.
- * @run main/manual=yesno DialogType
+ * @run main/manual DialogType
  */
 
-import java.awt.print.*;
-import javax.print.attribute.*;
-import javax.print.attribute.standard.*;
-
 public class DialogType {
 
-  static String[] instructions = {
-     "This test assumes and requires that you have a printer installed",
-     "It verifies that the dialogs behave properly when using new API",
-     "to optionally select a native dialog where one is present.",
-     "Two dialogs are shown in succession.",
-     "The test passes as long as no exceptions are thrown, *AND*",
-     "if running on Windows only, the first dialog is a native windows",
-     "control which differs in appearance from the second dialog",
-     ""
-  };
+    static String instruction = """
+            This test assumes and requires that you have a printer installed.
+            It verifies that the dialogs behave properly when using new API
+            to optionally select a native dialog where one is present.
+            Two dialogs are shown in succession.
+            The test passes as long as no exceptions are thrown, *AND*,
+            if running on Windows only, the first dialog is a native windows
+            control which differs in appearance from the second dialog.
+            Note: You can either press 'ESCAPE' button or click on the 'Cancel'
+            to close print dialog.
```

Comment:
Basically, the only thing that's required from the user is to close the dialogs shown by pressing Esc or clicking Cancel?

Once they're dismissed, the test could finish automatically. If it's the case, it's better to explain it in the instructions.

If an exception is thrown, the test will finish automatically.

---

It looks such a scenario is not covered by PassFailJFrame well: the instructions need to be shown for a short while before the test starts. After the user reads the instructions, the user clicks **Start** button. Then the instructions can be hidden or left on the screen (it depends on how long and complex the instructions are), and the test continues with its execution.

As soon as the test logic finishes, the test dismisses the instructions if they're still visible, and exits.

I know about at least three cases where a reduced functionality is required.

Reply:
Are you saying this test would be better with an enhancement to PassFailJFrame ?
But it is OK right, just more clunky than it could be .. ?